onnxruntime/core/session/inference_session.cc

```diff
@@ -1795,103 +1850,112 @@ Status InferenceSession::Run(const RunOptions& run_options,
   Status retval = Status::OK();
   const Env& env = Env::Default();
 
-  std::vector<IExecutionProvider*> exec_providers_to_stop;
-  exec_providers_to_stop.reserve(execution_providers_.NumProviders());
+  // Check if this Run() is simply going to be a CUDA Graph replay.
+  if (cached_cuda_execution_provider_for_cuda_graph_replay_ && cached_cuda_execution_provider_for_cuda_graph_replay_->IsGraphCaptured()) {
+      LOGS(*session_logger_, INFO) << "Replaying the captured CUDA Graph for this model";
+      ++current_num_runs_;
+      ORT_RETURN_IF_ERROR_SESSIONID_(cached_cuda_execution_provider_for_cuda_graph_replay_->GraphReplay());
+  } else {
+    if (cached_cuda_execution_provider_for_cuda_graph_replay_) {
+      LOGS(*session_logger_, INFO) << "Capturing the CUDA Graph for this model";
```

Comment:
Nit: Maybe we can move this check and log to the place in OnRunStart() in the CUDA EP. Other build flavors of ORT need not incur this runtime check cost ?

Reply:
> 1. Nit: Maybe we can move this check and log to the place in OnRunStart() in the CUDA EP. Other build flavors of ORT need not incur this runtime check cost ?

Good idea. Just updated it.

> 2. Nit: Can we avoid the `else` and the other formatting changes if just return Status::OK() in the if above ?

I also want to avoid `else` and the formatting changes. The issue with returning Status::OK() immediately is the code [here](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/session/inference_session.cc#L1895-L1921) will be skipped. What do you think of it?